aiokafka/consumer/group_coordinator.py

```diff
@@ -1202,46 +1217,67 @@ async def perform_group_join(self):
                 metadata = metadata.encode()
             group_protocol = (assignor.name, metadata)
             metadata_list.append(group_protocol)
+            # for KIP-394 we may have to send a second join request
+            try_join = True
+            while try_join:
+                try_join = False
+
+                if self._api_version < (0, 10, 1):
```

Comment:
nit: Could we abstract out this into a function?

Reply:
This depends on the KIP-394 decision, since this is tied to that. If we move that to another PR than this would revert to how we currently handle this in master.